src/main/java/org/spongepowered/api/world/World.java

```diff
@@ -262,6 +262,14 @@ default LocatableBlock getLocatableBlock(int x, int y, int z) {
     WorldBorder getWorldBorder();
 
     /**
+     * Sets the world border for this world.
+     * 
+     * @param border The world border to be used.
+     * @param cause The cause of the world border's change.
+     */
+    void setWorldBorder(WorldBorder border, Cause cause);
```

Comment:
Not sure if it's a good idea to allow replacing the world's `WorldBorder` because that's the one that is modified with commands and that is written to the disk. I think in this case it is better to let them modify it directly instead of replacing it.